module/VuFind/src/VuFind/Db/Row/Search.php

```diff
@@ -160,4 +171,236 @@ public function getUnsubscribeSecret(HMAC $hmac, $user)
         ];
         return $hmac->generate(array_keys($data), $data);
     }
+
+    /**
+     * Get identifier (returns null for an uninitialized or non-persisted object).
+     *
+     * @return ?int
+     */
+    public function getId(): ?int
+    {
+        return $this->id ?? null;
+    }
+
+    /**
+     * Get user.
+     *
+     * @return UserEntityInterface
+     */
+    public function getUser(): int
+    {
+        return $this->user_id
```

Comment:
This should just be `return $this->getDbServiceManager()->get(UserServiceInterface::class)->getUserById($this->user_id);` -- the method is not allowed to return null based on its return type.